### PR TITLE
The changes introduced.

### PR DESCRIPTION
**Pull request by vivekvpandya**
The changes introduced in this branch is required to make https://github.com/vivekvpandya/D8iOS to work with latest Drupal 8 release i.e RC1.
I have mentioned reason for almost every changes. But still if you find any doubt please let me know.
Some of change requires your attention because you may have better way to incorporate those changes.

Sincerely,
Vivek Pandya

### CODE REVIEW

File: DIOSComment.m [hunk 1]
```diff
@@ -25,9 +25,18 @@ + (void) createCommentWithParams:(NSDictionary*)params
 
     NSMutableDictionary *dict = [NSMutableDictionary new];
     NSString *href = [NSString stringWithFormat:@"%@/rest/type/%@/%@", [[[DIOSSession sharedSession] baseURL] absoluteString], @"comment", type];
-
-    NSString *relationKey = [NSString stringWithFormat:@"%@/rest/relation/comment/comment/entity_id", [[[DIOSSession sharedSession] baseURL] absoluteString]];
-    NSString *relationValue = [NSString stringWithFormat:@"%@/node/%@",[[[DIOSSession sharedSession] baseURL] absoluteString],relationID];
+// This is for older beta releases
```
Comment (kylebrowning) [inline DIOSComment.m]: Spacing is off.

File: DIOSComment.m [hunk 2]
```diff
@@ -25,9 +25,18 @@ + (void) createCommentWithParams:(NSDictionary*)params
...
+//    NSString *relationKey = [NSString stringWithFormat:@"%@/rest/relation/comment/comment/entity_id", [[[DIOSSession sharedSession] baseURL] absoluteString]];
```
Comment (kylebrowning) [inline DIOSComment.m]: Just removed these lines then.

File: DIOSSession.m [hunk 3]
```diff
@@ -56,9 +59,14 @@ - (void) sendRequestWithPath:(NSString*)path
         }
     }];
     if (_signRequests) {
+       
         NSURLCredential *credential = [NSURLCredential credentialWithUser:_basicAuthUsername password:_basicAuthPassword persistence:NSURLCredentialPersistenceNone];
         [operation setCredential:credential];
     }
+    else{
```
Comment (kylebrowning) [inline DIOSSession.m:65]: spacing after the else please.

File: DIOSSession.m [hunk 4]
```diff
@@ -67,13 +75,20 @@ - (NSMutableURLRequest *) requestWithMethod:(NSString *)method
...
     [self.requestSerializer setValue:@"application/json" forHTTPHeaderField:@"Accept"];
     [self.requestSerializer setValue:@"application/hal+json" forHTTPHeaderField:@"Content-Type"];
 
     if (self.signRequests) {
-        //doesntwork
+        
         [self.requestSerializer setAuthorizationHeaderFieldWithUsername:_basicAuthUsername password:_basicAuthPassword];
     }
+    else{
+        // If previously signRequest is set to YES then if temparoraly it has been set to NO then authorization header should be cleared
```
Comment (kylebrowning) [inline DIOSSession.m]: Can you rephrase this sentence or make it more complete? Im having a hard time reading it.

Comment (vivekvpandya) [inline DIOSSession.m]: I am explaining here ,
Consider the case where user is logged in to the Drupal via app i.e app is passing credentials on each request. But now if user has access to drupal through web and he/she changes the credentials there , in that case the app will receive 401 on request. So now app should no more supply credentials with request. I assume that boolean variable signRequest is there for this purpose. Or you can add one more helper method to remove the credentials from the shared session variable. I hope this clarifies.
